data/Builder.go

```diff
@@ -0,0 +1,110 @@
+package data
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+//Datum represent one data point
+type Datum map[string]interface{}
+
+//Dataset represents an array of data points
+type Dataset []Datum
```

Comment:
Cool - yeah I spent ages coming up with crap names so just left it like this but happy to change it .... did remember one thing that I think pushed me down this path though `data.Data` would be how its referenced